cmd/pebble/cmd_run.go

```diff
@@ -152,10 +160,13 @@ func runDaemon(rcmd *cmdRun, ch chan os.Signal) error {
 		tic = time.NewTicker(checkRunningConditionsRetryDelay)
 		checkTicker = tic.C
 	}
-
 	d.Version = cmd.Version
 	d.Start()
 
+	if rcmd.Dry {
+		return nil
+	}
```

Comment:
This seems a bit of an odd way to wire it up: by passing the flag through to the `Daemon`, then calling `Daemon.Start()`, but having it do nothing if dry is set. Why not just check `rcmd.Dry` after calling `Init`?

In fact, why not avoid passing the dry flag down at all? We could add a new `Daemon.Validate()` function which just checks `d.overlord.ServiceManager().Plan()`, and then call that directly from `runDaemon` if `rcmd.Dry` is set (and then exit). Otherwise call `Init()` (which could be updated to call `Validate`).

Note that this PR also has the side effect of making `pebble run` exit immediately if the existing plan is invalid. I think that's a good thing, but it is a behavioural change, so should probably be noted explicitly in the PR description.